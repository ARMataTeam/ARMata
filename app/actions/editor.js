// @flow
export const SET_TEMPLATE = 'SET_TEMPLATE';
export const MARK_EDITED = 'MARK_EDITED';
export const MARK_IDLE = 'MARK_IDLE';
export const MARK_SAVED = 'MARK_SAVED';

type actionType = {
  type: string
};

export function setNewTemplate(template: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: SET_TEMPLATE, template });
  };
}

export function markEdited() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_EDITED });
  };
}

export function markIdle() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_IDLE });
  };
}

export function markSaved() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_SAVED });
  };
}
