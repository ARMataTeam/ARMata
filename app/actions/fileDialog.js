export const OPEN_FILE = 'OPEN_FILE';

export function open(selectedFilename) {
  return {
    type: OPEN_FILE, selectedFilename
  };
}
