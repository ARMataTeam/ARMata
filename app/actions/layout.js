// @flow
import { ProgressInfo } from 'electron-updater';

export const CHANGE_LAYOUT = 'CHANGE_LAYOUT';
export const CHANGE_VIEW = 'CHANGE_VIEW';
export const CLEAR_ERRORS = 'CLEAR_ERRORS';
export const ERROR = 'ERROR';
export const ALERT = 'ALERT';
export const PROGRESS = 'PROGRESS';
export const REPORT_ERROR = 'REPORT_ERROR';
export const OPEN_SETTINGS = 'OPEN_SETTINGS';
export const CLOSE_SETTINGS = 'CLOSE_SETTINGS';
export const OPEN_NODE_WINDOW = 'OPEN_NODE_WINDOW';
export const CLOSE_NODE_WINDOW = 'CLOSE_NODE_WINDOW';
export const OPEN_WINDOW = 'OPEN_WINDOW';
export const CLOSE_WINDOW = 'CLOSE_WINDOW';
export const OPEN_VISUALIZATION = 'OPEN_VISUALIZATION';
export const OPEN_TOOLBOX = 'OPEN_TOOLBOX';
export const CLOSE_TOOLBOX = 'CLOSE_TOOLBOX';
export const TOGGLE_PHYSICS = 'TOGGLE_PHYSICS';
export const ADD_RESOURCE = 'ADD_RESOURCE';
export const DELETE_RESOURCE = 'DELETE_RESOURCE';

type actionType = {
  type: string
};

type changeViewType = {
  type: string,
  view: string
};

export function toggleHierarchicalLayout() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: CHANGE_LAYOUT });
  };
}

export function togglePhysics() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: TOGGLE_PHYSICS });
  };
}

export function changeView(view: string) {
  return (dispatch: (action: changeViewType) => void) => {
    dispatch({ type: CHANGE_VIEW, view });
  };
}

export function dispatchButtonClick(action: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: action });
    dispatch({ type: CLEAR_ERRORS });
  };
}

export function openSettings() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_SETTINGS });
  };
}

export function openNodeWindow(nodes: Array<string>) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_NODE_WINDOW, nodes });
  };
}

export function openVisualization() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_VISUALIZATION });
  };
}

export function openToolbox() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_TOOLBOX });
  };
}

export function closeToolbox() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: CLOSE_TOOLBOX });
  };
}

export function error(errorMessage: string, title: string = 'Error occured') {
  const buttons = [{ label: 'Got it', action: CLEAR_ERRORS }, { label: 'Report', action: REPORT_ERROR }];

  return {
    type: ERROR, message: errorMessage, buttons, title
  };
}

export function alert(alertMessage: string, buttons: Object[] = []) {
  return {
    type: ALERT, message: alertMessage, buttons
  };
}

export function notifyProgress(progress: ProgressInfo) {
  return {
    type: PROGRESS, progress
  };
}

export function clearErrors() {
  return {
    type: CLEAR_ERRORS
  };
}

export function openWindow(name: string, title: string, content: string) {
  return {
    type: OPEN_WINDOW, windowName: name, window: { title, content }
  };
}

export function closeWindow() {
  return {
    type: CLOSE_WINDOW
  };
}

export function addResource(resourceType: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: ADD_RESOURCE, resourceType });
  };
}


export function deleteResource(id: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: DELETE_RESOURCE, id });
  };
}