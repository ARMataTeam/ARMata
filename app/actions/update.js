// @flow
export const CHECKING_FOR_UPDATE = 'CHECKINF_FOR_UPDATE';
export const UPDATE_AVAILABLE = 'UPDATE_AVAILABLE';
export const UPDATE_NOT_AVAILABLE = 'UPDATE_NOT_AVAILABLE';

type actionType = {
  type: string
};

export function checkingForUpdate() {
  return { type: CHECKING_FOR_UPDATE };
}

export function updateAvailable() {
  return { type: UPDATE_AVAILABLE };
}

export function updateNotAvailable() {
  return { type: UPDATE_NOT_AVAILABLE };
}
