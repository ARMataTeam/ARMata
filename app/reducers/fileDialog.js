// @flow
import { OPEN_FILE } from '../actions/fileDialog';
import TemplateParser from '../parsers/templateParser';
import { Template } from '../types/template';

const stripJsonComments = require('strip-json-comments');

type actionType = {
  type: string
};

type fileDialogStateType = {
  selectedFilename: string,
  fileData: Template
};

const initialState = {
  selectedFilename: '',
  fileData: {
    schema: '',
    contentVersion: '',
    variables: [],
    outputs: [],
    resources: [],
    parameters: []
  }
};

export default function fileDialog(state: fileDialogStateType = initialState, action: actionType) {
  switch (action.type) {
    case OPEN_FILE: {
      const rawData = stripJsonComments(removeBOM(action.data));
      const json = JSON.parse(rawData);
      const templateParser = new TemplateParser(json);
      const parsedTemplate = templateParser.parseTemplate();

      TemplateParser.normalizeNames(parsedTemplate);

      return Object.assign({}, state, {
        selectedFilename: action.selectedFilename,
        fileData: parsedTemplate,
        hierarchicalLayout: false
      });
    }
    default: {
      return state;
    }
  }
}

function removeBOM(data: string) {
  let clearedData = data;
  if (clearedData.charCodeAt(0) === 0xFEFF) {
    clearedData = clearedData.slice(1);
  }

  return clearedData;
}
