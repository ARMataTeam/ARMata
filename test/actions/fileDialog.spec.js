import * as actions from '../../app/actions/fileDialog';

describe('actions', () => {
  it('should open should create open action', () => {
    expect(actions.open()).toMatchSnapshot();
  });
});
