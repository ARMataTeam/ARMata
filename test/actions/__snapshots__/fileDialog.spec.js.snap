// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should open should create open action 1`] = `
Object {
  "data": undefined,
  "selectedFilename": undefined,
  "type": "OPEN_FILE",
}
`;
